src/main/java/org/apache/commons/compress/compressors/gzip/GzipParameters.java

```diff
@@ -118,4 +120,28 @@ public int getOperatingSystem() {
     public void setOperatingSystem(final int operatingSystem) {
         this.operatingSystem = operatingSystem;
     }
+
+    /**
+     * Gets size of the buffer used to retrieve compressed data.
+     * @return The size of the buffer used to retrieve compressed data.
+     * @since 1.21
+     * @see #setBufferSize(int)
+     */
+    public int getBufferSize() {
+        return this.bufferSize;
+    }
+
+    /**
+     * Sets size of the buffer used to retrieve compressed data from
+     * {@link Deflater} and write to underlying {@link OutputStream}.
+     * 
+     * @param bufferSize the bufferSize to set. Must be a positive value.
+     * @since 1.21
+     */
+    public void setBufferSize(int bufferSize) {
+        if (bufferSize <= 0) {
```

Comment:
Do we need a high limit check for `bufferSize` here?

Reply:
I personally do not see the value, but can add if desired. 
There is certainly some threshold where larger values are counter productive, but my opinion is consumers can determine that on their own.